helm/templates/logger-network-policy.yaml

```diff
@@ -18,15 +20,29 @@ spec:
     ports:
     - protocol: TCP
       port: 8500
-  {{- if not $.Values.minio.metricsLogger.withRedis }}
+  {{- if or (not $.Values.minio.metricsLogger.withRedis) (eq $.Values.minio.metricsLogger.LOG_PROVIDER "elasticsearch") (eq $.Values.minio.metricsLogger.LOG_PROVIDER "s3") }}
   egress:
   - to:
     - ipBlock:
         cidr: 0.0.0.0/0
     ports:
     - protocol: UDP
       port: 53
+    {{- if not $.Values.minio.metricsLogger.withRedis }}
     - protocol: TCP
       port: {{ $.Values.minio.metricsLogger.REDIS_PORT }}
+    {{- end }}
+    {{- if (eq $.Values.minio.metricsLogger.LOG_PROVIDER "elasticsearch") }}
+    - protocol: TCP
+      port: {{ $.Values.minio.metricsLogger.ES_PORT }}
```

Comment:
it should also handle possibility of multiple hosts with different ports in ES_HOSTS

Reply:
there are some more edge cases which I think are not handled by your algorythm.. I really think we should not parse URLs ourselves, maybe find a helm function to do it? Otherwise, we should do it in the operator Python code..

Following cases I think will not be handled correctly (but there might be more because URL parsing can be hard..):

* `ES_HOSTS=http://example.com,https://example.com`
  * expected ports: 80, 443
  * actual: port will not be parsed as it doesn't appear in the URL
* `ES_HOSTS=Http://example.com:1234,HTTPS://example.com:4321`
  * expected ports: 1234, 4321
  * actual: ports will not be parsed due to uppercase characters in scheme
* `ES_HOSTS=HTTP://example.com`
  * expected port: 80
  * actual port: 443 (because of uppercase HTTP)
* `ES_HOSTS=://example.com:1234`
  * expected port: 1234
  * actual: will not be parsed because of missing http prefix (:// is a valid prefix for http)